pkg/logging/logging.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2021 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/go-logr/glogr"
+	"github.com/go-logr/logr"
+)
+
+// A logger embeds logr.Logger
+type Logger struct {
```

Comment:
this can be a private type.  New() should return `logr.Logger`

Reply:
but looks like if we return logr.Logger we cant reach "custom Error()" and other functions (deleteErrorFile...)?

Reply:
we can implement LogSink  interface and just call same functions again maybe but this still can't solve DeleteErrorFile() function?

Reply:
I forgot about the custom methods.  That's fine.